dataSourceImpl/src/main/java/model/PreferencesRepository.kt

```diff
@@ -0,0 +1,19 @@
+package model
+
+import android.content.Context
+import android.content.SharedPreferences
+import com.panhuk.core.BuildConfig
+
+class PreferencesRepository(context: Context) {
+    private var sharedPreferences: SharedPreferences =
+        context.getSharedPreferences(BuildConfig.APP_PREFERENCES, Context.MODE_PRIVATE)
```

Comment:
It's better to use [jetpack datastore](https://developer.android.com/topic/libraries/architecture/datastore?gclid=Cj0KCQiAhf2MBhDNARIsAKXU5GT-EIWjFhULziJ-E3JW_Msz-wx3w4q3U1NTWOGBNH7o7aba0SI3KWoaAs3pEALw_wcB&gclsrc=aw.ds) instead of shared preferences